src/lib/FlightTasks/tasks/Utility/ObstacleAvoidance.cpp

```diff
@@ -99,7 +103,9 @@ void ObstacleAvoidance::injectAvoidanceSetpoints(Vector3f &pos_sp, Vector3f &vel
 	const bool avoidance_point_valid =
 		_sub_vehicle_trajectory_waypoint->get().waypoints[vehicle_trajectory_waypoint_s::POINT_0].point_valid == true;
 
-	if (avoidance_data_timeout || !avoidance_point_valid) {
+	_avoidance_point_valid_hysteresis.set_state_and_update(!avoidance_point_valid, hrt_absolute_time());
```

Comment:
I don't really like that the `_avoidance_point_valid_hysteresis` has the opposite meaning of `avoidance_point_valid`. Please put the `!` in the `if` check below or rename the variable.